### DISCUSSION TITLE
Error when running python -m omnigibson.examples.learning.navigation_policy_demo

### DISCUSSION THREAD

**Discussion by ge79puv** [StanfordVL/BEHAVIOR-1K#957] [Q&A] [answered]
Hey there, 

I just used Isaac Sim 4.1.0 and the new released OmniGibson v 1.1.1. I tried to run the demo of RL. However, there is an error: 

![image](https://github.com/user-attachments/assets/0f0d8471-3635-41e9-ab2a-b39f00d0d15c)

Could you help me to solve this issue?

Best regards

**Comment by ChengshuLi** [comment]
Could you let us know the version of `gymnasium`?

**Comment by ge79puv** [comment]
0.29.1

**Comment by ChengshuLi** [comment]
Hmm I am actually not able to reproduce your issue. The example ran okay on my side (also OG 1.1.1 and Isaac Sim 4.1). Maybe you can take a closer look at `obs.spaces`? If it's not `OrderedDict`, what is it?

For your reference, 

```
>>> import gymnasium
>>> gymnasium
<module 'gymnasium' from '/path/to/isaac/isaac-sim-4.1.0/exts/omni.isaac.ml_archive/pip_prebundle/gymnasium/__init__.py'>
>>> gymnasium.__version__
'0.28.1'
```

My `gymnasium` version is lower. Are you using `gymnasium` shipped with Isaac?

**Comment by ge79puv** [comment]
I downgrade my gymnasium version, now it works. Thanks a lot!

**Comment by ge79puv** [comment]
Hi,

this omnigibson.examples.learning.navigation_policy_demo sometimes runs fine, but sometimes it suddenly terminates after running for a while. I have not modified any source code, and the error message is as follows: 
![image](https://github.com/user-attachments/assets/03af6612-0896-40b9-9ad9-1cbd8ce85fc1)
I cannot understand why the tensors are loaded on two gpus. And this termination is uncertain. Sometimes this error appears at the beginning, and sometimes it appears after running for a period of time, which makes me unable to complete the entire RL training.

Best regards

**Reply by ChengshuLi** [reply, marked as answer]
I think the problem is that the neural network weights sit on one GPU and the observation you get from the environment (e.g. a RGB image tensor) sits on another GPU. You can try to make the policy model to sit on the same GPU as OmniGibson.

**Reply by ge79puv** [reply]
Yes, you are right. I manually set the device of PPO model, then this error never appears. Thanks for your help!

**Comment by ge79puv** [comment]
Another interesting phenomenon is that, when run a demo, after a while, the demo will terminate without any specific instruction, just a "Fatal Python error: Segmentation fault". Do you know how can I fix that?

**Reply by ChengshuLi** [reply]
I think this should be safe to ignore. This happens when we didn't gracefully terminate Isaac (by calling `og.shutdown()`). We will make sure to add proper shutdown for all our examples in the next release. Thanks!
